Tests/Recurly/ShippingAddress_Test.php

```diff
@@ -61,4 +61,18 @@ public function testUpdateShippingAddress() {
     }
   }
 
+  public function testDeleteShippingAddress() {
+    $this->client->addResponse('GET', '/accounts/abcdef1234567890/shipping_addresses', 'shipping_addresses/index-200.xml');
+    $this->client->addResponse('DELETE', 'https://api.recurly.com/v2/accounts/abcdef1234567890/shipping_addresses/1234567', 'shipping_addresses/destroy-204.xml');
+
+    $shipping_addresses = Recurly_ShippingAddressList::get('abcdef1234567890', null, $this->client);
+
+    foreach ($shipping_addresses as $shipping_address) {
```

Comment:
I think it's safe to just take the first one?

```php
shipping_addresses[0]->delete();
```

If you want to make it more future proof (optional). You can use the shipping address to setup the stub. Something like:

```php
$address = $shipping_addresses[0];
$this->client->addResponse('DELETE', 'https://api.recurly.com/v2/accounts/abcdef1234567890/shipping_addresses/' . $address->id, 'shipping_addresses/destroy-204.xml');
```

Reply:
`shipping_addresses[0]` yields `Cannot use object of type Recurly_ShippingAddressList as array`, which may or may have not led to a StackOverflow wormhole.